### PR TITLE
Cover Customers Update/Create pages

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A
|-----------------|-----
| Branch?         | 2.0 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| License         | MIT

Screenshot 2024-07-03 at 10 42 31

### CODE REVIEW

File: features/admin/user/managing_customers/being_able_to_create_account_for_existing_customer.feature
Comment (TheMilek) [inline features/admin/user/managing_customers/being_able_to_create_account_for_existing_customer.feature:1]: Deleted, as we abandoned the "can log in?" toggle

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php [hunk 1]
```diff
@@ -54,15 +56,15 @@ public function iWantToCreateANewCustomer()
      */
     public function iSpecifyItsFirstNameAs($name)
```
Comment (Wojdylak) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php]:
```suggestion
    public function iSpecifyItsFirstNameAs($name): void
```

File: src/Sylius/Behat/Element/Admin/Customer/FormElement.php [hunk 2]
```diff
@@ -84,6 +68,41 @@ public function verifyUser(): void
         $this->getDocument()->checkField('Verified');
     }
 
+    public function specifyFirstName(string $name): void
+    {
+        $this->getDocument()->fillField('First name', $name);
```
Comment (Wojdylak) [inline src/Sylius/Behat/Element/Admin/Customer/FormElement.php]:
```suggestion
        $this->getElement('first_name')->setValue($name);
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/customer/create.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+        'sylius_admin.customer.create.content.form.sections':
+            details:
+                template: '@SyliusAdmin/customer/form/sections/details.html.twig'
+            extra_information:
+                template: '@SyliusAdmin/customer/form/sections/extra_information.html.twig'
+            account_credentials:
+                template: '@SyliusAdmin/customer/form/sections/account_credentials.html.twig'
+                
+        'sylius_admin.customer.create.content.form.sections.account_credentials':
```
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/customer/create.yaml]: Could you keep the same order like above.

File: src/Sylius/Behat/Element/Admin/Customer/FormElement.php [hunk 4]
```diff
@@ -97,7 +116,7 @@ protected function getDefinedElements(): array
             'first_name' => '#sylius_admin_customer_firstName',
             'group' => '#sylius_admin_customer_group',
             'last_name' => '#sylius_admin_customer_lastName',
-            'password' => '#sylius_admin_customer_user_password',
+            'password' => '#sylius_admin_customer_user_plainPassword',
```
Comment (Wojdylak) [inline src/Sylius/Behat/Element/Admin/Customer/FormElement.php]: Could you use the sylius_form_test attribute?

File: src/Sylius/Bundle/CoreBundle/Form/Extension/CustomerTypeExtension.php [hunk 5]
```diff
@@ -1,33 +0,0 @@
...
-
-namespace Sylius\Bundle\CoreBundle\Form\Extension;
-
-use Sylius\Bundle\CoreBundle\Form\EventSubscriber\AddUserFormSubscriber;
-use Sylius\Bundle\CoreBundle\Form\Type\User\ShopUserType;
-use Sylius\Bundle\CustomerBundle\Form\Type\CustomerType;
-use Symfony\Component\Form\AbstractTypeExtension;
-use Symfony\Component\Form\FormBuilderInterface;
-
-final class CustomerTypeExtension extends AbstractTypeExtension
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/Extension/CustomerTypeExtension.php:30]: Deprecate this extensions in 1.14

File: src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddUserFormSubscriber.php [hunk 6]
```diff
@@ -1,65 +0,0 @@
...
-
-use Sylius\Component\User\Model\UserAwareInterface;
-use Symfony\Component\EventDispatcher\EventSubscriberInterface;
-use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
-use Symfony\Component\Form\FormEvent;
-use Symfony\Component\Form\FormEvents;
-use Symfony\Component\Validator\Constraints\Valid;
-use Webmozart\Assert\Assert;
-
-final class AddUserFormSubscriber implements EventSubscriberInterface
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddUserFormSubscriber.php:32]: Deprecate this subscriber in 1.14

**Review by GSadee — DISMISSED** [review]

File: src/Sylius/Bundle/AdminBundle/Form/Type/CustomerType.php [hunk 7]
```diff
@@ -13,11 +13,39 @@
...
+            Assert::isInstanceOf($data, UserAwareInterface::class);
+
+            if (null === $data->getUser()->getPlainPassword() && null === $data->getUser()->getId()) {
+                $data->setUser(null);
+                $event->setData($data);
+
+                $form->remove('user');
+                $form->add('user', ShopUserType::class, ['constraints' => [new Valid()]]);
+            }
+        });
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Form/Type/CustomerType.php]: Maybe extract this logic outside so it'll be easier to customize/remove if a need arises

File: src/Sylius/Bundle/AdminBundle/templates/customer/form/sections/account_credentials.html.twig [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    <div class="card">
+        <div class="card-header">
+            <div class="card-title">
+                {{ 'sylius.ui.account_credentials'|trans }}
+            </div>
+        </div>
+        <div class="card-body">
+            <div class="alert alert-info">
+                {{ 'sylius.ui.account_credentials_details'|trans }}
+            </div>
```
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/templates/customer/form/sections/account_credentials.html.twig:11]: I think it should only be visible if the shop user has not been created.

**Review by Wojdylak — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Pull request merged**